File: column_cat.py
import os
import csv
from tqdm import tqdm


def listdir_nohidden(path):
    for f in os.listdir(path):
        if not f.startswith('.'):
            yield f


directory = '/Users/dolteanu/local_documents/Coding/MSc/Data/letter calls'
files = list(listdir_nohidden(directory))
outfile = open(
    "/Users/dolteanu/local_documents/Coding/MSc/Output/calls.csv", 'w')

writer = csv.writer(outfile, delimiter=',')

writer.writerow([''] + files)
line_data = []
for filename in tqdm(files):
    print(filename)
    file = open(directory+'/'+filename, 'r')
    lines = tqdm(file.readlines())
    for x, line in enumerate(lines):
        if x != 0:
            line_strip = line.strip('\n').split(' ')
            line_elements = line_strip[0].split('\t')
            first_calls = line_elements[1]
            if filename == os.listdir(directory)[0]:
                line_data.append([line_elements[0]])

            line_data[x-1].append(first_calls)

writer.writerows(line_data)
outfile.close()

File: column_cat.r

# create code to ask user to input location of files, store in variable & pass var to setwd function
# ask user for any patterns to be removed from filename to match feature labels
setwd('/Users/dolteanu/local_documents/Coding/MSc/FinalProject/Data/letter calls/')
files<- list.files(path='./', pattern='.txt')
colname<-vector()
genotyping<-vector()
files[1]
name <- files[1]
name2 <-files[2]
samples <-read.table(name,sep = "\t",header=T)
samples2 <- read.table(name,header = T,sep = "\t")

write.table(samples[,2], file = "../../Output/calls.csv",row.names=samples[,1], col.names=name,sep=",")

write.table(c("\n",name,samples2[,2]),file = "../../Output/calls.csv", append=T, col.names=F,sep=",")


##once the second write.table works add it into the for loop 
# for (i in files) {
#   name<- toString(i)
#   sample <- read.csv(i)
# }
# write.
# c(" ",name)
